test/framework/integration_framework/fake_peer/fake_peer.cpp

```diff
@@ -55,191 +53,191 @@ static std::shared_ptr<shared_model::interface::Peer> createPeer(
-        ->createSignature(keypair_->publicKey(), bare_signature)
-        .match(
-            [&signature_with_pubkey](
-                iroha::expected::Value<
-                    std::unique_ptr<shared_model::interface::Signature>> &sig) {
-              signature_with_pubkey = std::move(sig.value);
-            },
-            [this](iroha::expected::Error<std::string> &reason) {
-              BOOST_THROW_EXCEPTION(std::runtime_error(
-                  "Cannot build signature: " + reason.error));
-            });
-    return signature_with_pubkey;
-  }
-
-  iroha::consensus::yac::VoteMessage FakePeer::makeVote(
-      const iroha::consensus::yac::YacHash &yac_hash) {
-    iroha::consensus::yac::YacHash my_yac_hash = yac_hash;
-    my_yac_hash.block_signature = makeSignature(
-        shared_model::crypto::Blob(yac_hash.vote_hashes.block_hash));
-    return yac_crypto_->getVote(my_yac_hash);
-  }
-
-  void FakePeer::sendYacState(
-      const std::vector<iroha::consensus::yac::VoteMessage> &state) {
-    yac_transport_->sendState(*real_peer_, state);
-  }
-
-  void FakePeer::voteForTheSame(const FakePeer::YacMessagePtr &incoming_votes) {
-    using iroha::consensus::yac::VoteMessage;
-    log_->debug("Got a YAC state message with {} votes.",
-                incoming_votes->size());
-    if (incoming_votes->size() > 1) {
-      // TODO mboldyrev 24/10/2018 IR-1821: rework ignoring states for accepted
-      //                                    commits
-      log_->debug(
-          "Ignoring state with multiple votes, "
-          "because it probably refers to an accepted commit.");
-      return;
+
+    rxcpp::observable<FakePeer::OgProposalPtr>
+    FakePeer::get_og_proposals_observable() {
+      return og_network_notifier_->get_observable();
+    }
+
+    std::shared_ptr<shared_model::interface::Signature> FakePeer::makeSignature(
+        const shared_model::crypto::Blob &hash) const {
+      auto bare_signature =
+          shared_model::crypto::DefaultCryptoAlgorithmType::sign(hash,
+                                                                 *keypair_);
+      std::shared_ptr<shared_model::interface::Signature> signature_with_pubkey;
+      common_objects_factory_
+          ->createSignature(keypair_->publicKey(), bare_signature)
+          .match([&signature_with_pubkey](
+                     iroha::expected::Value<
+                         std::unique_ptr<shared_model::interface::Signature>> &
+                         sig) { signature_with_pubkey = std::move(sig.value); },
+                 [this](iroha::expected::Error<std::string> &reason) {
+                   BOOST_THROW_EXCEPTION(std::runtime_error(
+                       "Cannot build signature: " + reason.error));
+                 });
+      return signature_with_pubkey;
+    }
+
+    iroha::consensus::yac::VoteMessage FakePeer::makeVote(
+        const iroha::consensus::yac::YacHash &yac_hash) {
+      iroha::consensus::yac::YacHash my_yac_hash = yac_hash;
```

Comment:
Yes, in this case passing by value makes some sense, for example if we would call `makeVote` with a `YacHash` rvalue, then the move c-tor will be used instead of copy c-tor.